usr/src/uts/common/fs/zfs/sys/dnode.h

```diff
@@ -136,7 +136,13 @@ typedef struct dnode_phys {
 	uint8_t dn_flags;		/* DNODE_FLAG_* */
 	uint16_t dn_datablkszsec;	/* data block size in 512b sectors */
 	uint16_t dn_bonuslen;		/* length of dn_bonus */
-	uint8_t dn_pad2[4];
+	/*
+	 * crypto is ready to encrypt dnodes, but until large-dnode
+	 * commit is merged, dn_extra_slots is left here as a
+	 * place-holder, but assumed to be 0 at all times.
+	 */
```

Comment:
Are you thinking that we will merge large-dnode support to illumos before crypto, in which case this comment would go away?  If not, I think it's OK to keep this code here but I'd suggest rewording the comment:
`dn_extra_slots is a placeholder for a feature in other ZFS implementations.  In this implementation, its value is always 0.  We declare it here to ensure it isn't used for a different purpose, and to improve code portability with implementations which support extra dnode slots.`